src/parser/core/Analyzer.ts

```diff
@@ -11,88 +11,22 @@ import { EventType, MappedEvent } from './Events';
 export { SELECTED_PLAYER, SELECTED_PLAYER_PET };
 export type Options = _Options
 
-const EVENT_LISTENER_REGEX = /on_((by|to)Player(Pet)?_)?(.+)/;
-
-/**
- * Get a list of all methods of all classes in the prototype chain until this
- * class. Orginal source: https://stackoverflow.com/a/40577337/684353
- * @param {object} obj A class instance
- * @returns {Set<any>}
- */
-function getAllChildMethods(obj: object) {
-  // Set is required here to avoid duplicate methods (if a class extends another it might override the same method)
-  const methods = new Set<string>();
-  // eslint-disable-next-line no-cond-assign
-  while ((obj = Object.getPrototypeOf(obj)) && obj !== Analyzer.prototype) {
-    const keys = Object.getOwnPropertyNames(obj).concat(
-      (Object.getOwnPropertySymbols(obj) as unknown) as string[],
-    );
-    keys.forEach(k => methods.add(k));
-  }
-  return methods;
-}
-function addLegacyEventListenerSupport(object: Analyzer) {
-  const methods = getAllChildMethods(object);
-  let hasLegacyEventListener = false;
-  // Check for any methods that match the old magic method names and connect them to the new event listeners
-  // Based on https://github.com/xivanalysis/xivanalysis/blob/a24b9c0ed8b341cbb8bd8144a772e95a08541f8d/src/parser/core/Module.js#L51
-  methods.forEach(name => {
-    const match = EVENT_LISTENER_REGEX.exec(name);
-    if (!match) {
-      return;
-    }
-    const [listener, , playerFilter, pet, eventType] = match;
-    const filter = new EventFilter(eventType as EventType);
-
-    // This only shows available filters used by the legacy method.
-    // For a full list of supported properties see the core CombatLogParser
-    let by = 0;
-    if (playerFilter === 'by' && !pet) {
-      by = by | SELECTED_PLAYER;
-    }
-    if (playerFilter === 'by' && pet) {
-      by = by | SELECTED_PLAYER_PET;
-    }
-    filter.by(by);
-    let to = 0;
-    if (playerFilter === 'to' && !pet) {
-      to = to | SELECTED_PLAYER;
-    }
-    if (playerFilter === 'to' && pet) {
-      to = to | SELECTED_PLAYER_PET;
-    }
-    filter.to(to);
-
-    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-    // @ts-ignore
-    object.addEventListener(filter, object[listener]);
-    hasLegacyEventListener = true;
-  });
-  object.hasLegacyEventListener = hasLegacyEventListener;
-}
-
 class Analyzer extends EventSubscriber {
-  hasLegacyEventListener = false;
 
   /**
    * Called when the parser finished initializing; after all required
    * dependencies are loaded, normalizers have ran and combatants were
    * initialized. Use this method to toggle the module on/off based on having
    * items equipped, talents selected, etc.
    */
+  // eslint-disable-next-line @typescript-eslint/no-useless-constructor
```

Comment:
Is there a reason the constructor cant be removed?

Reply:
it can be removed; I left it there as a "documentation" as that comment prior is probably a starting point for new contributors extending the analyzer module, and didn't wanna leave that comment floating (see other methods being defined as empty with exceptions for example)